app/Services/DependencyService.php

```diff
@@ -4,12 +4,46 @@
 
 use App\Models\Dependency\Dependency;
 use App\Models\User\User;
+use Cache;
 use Carbon\Carbon;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\Log;
+use InvalidArgumentException;
 use LogicException;
 
 class DependencyService
 {
+    /**
+     * Given a dependency, fetch the cached version or cache if it we need to.
+     */
+    public static function fetchDependencyDataById(int $id): array
+    {
+        return self::fetchDependencyData(Dependency::findOrFail($id));
+    }
+
+    private static function fetchDependencyData(Dependency $dependency): array
+    {
+        return Cache::rememberForever(
+            self::getDependencyCacheKey($dependency),
+            function () use ($dependency) {
+                $response = app()->call(sprintf('App\\Http\\Controllers\\%s', $dependency->action));
```

Comment:
Personal preference, I'd use the string interpolation method instead of sprintf :D

Reply:
Done!